client.go

```diff
@@ -0,0 +1,47 @@
+package chat
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+)
+
+// Client represents a chat client
+type Client struct {
+	connection net.Conn
+}
+
+// ConnectClient returns a new client with a connection to the server
+func ConnectClient(addr string) (*Client, error) {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return &Client{}, err
```

Comment:
When we return a pointer, the convention is to return `nil` in the error case. (We would like to do that when we return structs too, but structs can't be nil, so we have to return an empty struct.)